### PR TITLE
Features/navigate select options

### PR DESCRIPTION
**Pull request by nfiniteset**

### CODE REVIEW

**Comment by nfiniteset** [comment]
Related to #46

**Comment by camwest** [comment]
When testing in react-playground I found some odd behaviour:

* When I tab to the component and press down once nothing happens
  * Expected: The menu should open
* When I press down a three times it opens
* Same behaviour with up arrow key or mouse clicking
* Clicking outside doesn't close the menu anymore (is there an test for this?)
* Position of the popover seems broken (see screenshot)

screenshot 2017-01-30 10 43 18

**Comment by camwest** [comment]
High Level Feedback:

* the selectedIndex should be visually highlighted and distinct from the focusIndex when the menu is open
* pressing down when the menu is closed should not change the focusIndex when there is a selectedIndex, it should ONLY change the open/closed status
  * check out the spec detailed here: https://github.com/orion-ui/orion/issues/42
     * down or up arrow when closed and having a selectedIndex should set the focusedIndex to the selectedIndex

**Comment by aneyzberg** [comment]
@camwest @nfiniteset and I have resolved the high level concerns you raised above. please take a look when you have a moment.

**Comment by camwest** [comment]
@aneyzberg OK it looks GREAT functionally. Let me review the source tonight and I'll get back to you.

File: packages/storybook/stories/select/selected-index.js [hunk 1]
```diff
@@ -19,17 +19,23 @@ import React from 'react';
 import { boolean, number } from '@kadira/storybook-addon-knobs';
 
 import SourceViewer from '../../components/source_viewer';
+import Example from '../../components/example';
+import { Select } from '../../../react/lib/2016-12-01';
 
 module.exports = function selectedIndex() {
   const props = {
     open: boolean('Open', false),
```
Comment (camwest) [inline packages/storybook/stories/select/selected-index.js]: Can you default this to open?

File: packages/storybook/stories/select/expanded.js [hunk 2]
```diff
@@ -85,7 +85,7 @@ angular.module('app', [])
   return (
     <div>
       <Example minHeight={100}>
-        <Select options={options} open={props.open} />
+        <Select options={options} selectedIndex={1} open={props.open} />
```
Comment (camwest) [inline packages/storybook/stories/select/expanded.js]: Can you remove this selectedIndex? This view state should show the menu open without a selection.

File: packages/angular-playground/src/index.html [hunk 3]
```diff
@@ -24,21 +24,18 @@
...
-    </style>
   </head>
   <body ng-app="app">
     <h1>Angular Playground</h1>
     <hr />
     <div ng-controller="Controller as ctrl">
       <h2>Buttons</h2>
       <a href="#" ng-click="ctrl.toggleDisability()">Toggle Button Disability</a> |
-      <a href="#" ng-click="ctrl.embiggen()">Make all buttons big</a>
+      <orion-select on-change="ctrl.setSize(event)" options="sizes" ng-model="buttonSize"></orion-select>
```
Comment (camwest) [inline packages/angular-playground/src/index.html]: What is the purpose of ng-model here?

I'd like to avoid using two-way binding at all.

**Comment by camwest** [comment]
Also, Is key a required property for an option? If so- can you update all source examples to show the key?

**Comment by aneyzberg** [comment]
@camwest I believe @nfiniteset and I addressed the changes mentioned above. Take a look when you have a moment.

**Pull request merged**